### PR TITLE
[JBWS-4429] NPE when SOAP request with mismatched targetNameSpace is received

### PR DESCRIPTION
**Pull request by gaol**
Issue: https://issues.redhat.com/browse/JBWS-4429 

This PR is based on https://github.com/jbossws/jbossws-cxf/pull/530 and adds a test case for it.

### CODE REVIEW

**Comment by gaol** [comment]
@jimma would you please review ? thanks

**Comment by jimma** [comment]
Thanks @lvydra  and @gaol !

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/interceptor/HandlerAuthInterceptor.java [hunk 1]
```diff
@@ -144,6 +144,10 @@ protected void checkAuthorization(MessageContext ctx)
             SecurityContext secCtx = message.get(SecurityContext.class);
             BindingOperationInfo bop = exchange.getBindingOperationInfo();
             MethodDispatcher md = (MethodDispatcher) exchange.getService().get(MethodDispatcher.class.getName());
+            if (bop == null)
+            {
+               throw MESSAGES.missingBindingOperationForAuthorization();
+            }
```
Comment (asoldano) [inline modules/server/src/main/java/org/jboss/wsf/stack/cxf/interceptor/HandlerAuthInterceptor.java:150]: Perhaps we can move this stuff by 1 line? @jimma

Comment (jimma) [reply modules/server/src/main/java/org/jboss/wsf/stack/cxf/interceptor/HandlerAuthInterceptor.java:150]: Yes @asoldano . This is better. I'll get this fixed in the next commit.

**Pull request merged**